### PR TITLE
[MINOR]: Remove unecessary orderings from the final plan

### PR DESCRIPTION
**Pull request by mustafasrepo**
## Which issue does this PR close?

Closes #.

## Rationale for this change
In some of the plans, there are unnecessary `SortExec`, where ordered fields are not used (exist at the final plan.) This PR remove `SortExec`s for these cases from the plan.

### CODE REVIEW

**Comment by ozankabak** [comment]
(this doesn't exactly do the right thing -- we will update this today and re-open)

**Review by ozankabak — APPROVED** [review]
LGTM. @alamb, would appreciate a quick look from you

**Review by Dandandan — APPROVED** [review]

File: datafusion/physical-plan/src/insert.rs [hunk 1]
```diff
@@ -219,24 +219,17 @@ impl ExecutionPlan for FileSinkExec {
...
 
     fn required_input_ordering(&self) -> Vec<Option<Vec<PhysicalSortRequirement>>> {
-        // The input order is either exlicitly set (such as by a ListingTable),
-        // or require that the [FileSinkExec] gets the data in the order the
-        // input produced it (otherwise the optimizer may chose to reorder
-        // the input which could result in unintended / poor UX)
-        //
-        // More rationale:
-        // https://github.com/apache/arrow-datafusion/pull/6354#discussion_r1195284178
-        match &self.sort_order {
```
Comment (Dandandan) [inline datafusion/physical-plan/src/insert.rs:229]: We experienced some problems with this as well (added sorts), so happy to see this go :)

File: datafusion/sqllogictest/test_files/select.slt [hunk 2]
```diff
@@ -1013,6 +1013,29 @@ SortPreservingMergeExec: [c@3 ASC NULLS LAST]
...
+GROUP BY c2;
+----
+logical_plan
+Aggregate: groupBy=[[aggregate_test_100.c2]], aggr=[[COUNT(UInt8(1)) AS COUNT(*)]]
+--Projection: aggregate_test_100.c2
+----Sort: aggregate_test_100.c1 ASC NULLS LAST, aggregate_test_100.c2 ASC NULLS LAST
+------Projection: aggregate_test_100.c2, aggregate_test_100.c1
+--------TableScan: aggregate_test_100 projection=[c1, c2]
+physical_plan
+AggregateExec: mode=FinalPartitioned, gby=[c2@0 as c2], aggr=[COUNT(*)]
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/select.slt:1032]: 👍  it is nice to see no Sort!

**Review by alamb — APPROVED** [review]
Thank you @mustafasrepo and @Dandandan

**Pull request merged**